### ISSUE TITLE
`prefer-code-point` should not suggest a change if the number is used for a sum

### ISSUE THREAD

**Issue by BridgeAR** [sindresorhus/eslint-plugin-unicorn#3425]
Code points are indeed generally favorable, while a replacement should not happen if the numbers are used for calculations such as building a sum, since that changes the behavior.

```js
let hash = 0
const size = hashSource.length
for (let offset = 0; offset < size; offset++) {
  hash = (((hash * 31) | 0) + hashSource.charCodeAt(offset)) | 0
}
```

**Comment by sindresorhus** [comment]
Good catch, though the rename is actually *broken* here, not just different: `charCodeAt` and `codePointAt` only diverge at an astral char's high surrogate. For `'🦄'` (`0xD83E 0xDD84`, `length === 2`), `codePointAt(0)` returns `0x1F984` but the loop still visits the leftover low surrogate at index 1, giving a broken hybrid. A real code-point hash needs `for...of`.

But that's exactly why I'm keeping the report and only dropping the suggestion: `charCodeAt` in a `length` loop silently mishandles emoji, so most of the time it's a latent bug worth flagging, the auto-fix just can't fix it safely. Cases where it's intentional are a quick disable.

**Comment by BridgeAR** [comment]
Yes, in most situations it is a bug, agreed.

What I suggest is to try to detect what is done with the numbers :). If the value is used for a sum (or other calculations), the report could be skipped, since it is does something out of the norm.
